galsim/photon_array.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2012-2016 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""@file photon_array.py
+Implements the PhotonArray class describing a collection of photons incident on a detector.
+Also includes classes that modify PhotonArray objects in a number of ways.
+"""
+
+# Most of the functionality comes from the C++ layer
+from ._galsim import PhotonArray
+import numpy as np
+
+# Add on more methods in the python layer
+
+PhotonArray.__doc__ = """
+The PhotonArray class encapsulates the concept of a collection of photons incident on
+a detector.
+
+A PhotonArray object is not typically constructed directly by the user.  Rather, it is
+typically constructed as the return value of the `GSObject.shoot` method.
```

Comment:
This anticipates the change Josh is making on #824, where this will be a method of GSObject rather than SBProfile.  So I think the comment is wrong for this branch, but will be correct for the released 1.5.